lib/iphone-simulator-xcode-simctl.ts

```diff
@@ -116,11 +120,64 @@ export class XCodeSimctlSimulator extends IPhoneSimulatorNameGetter implements I
 	}
 
 	public printDeviceLog(deviceId: string, launchResult?: string): any {
-		return common.printDeviceLog(deviceId, launchResult);
+		let pid = "";
+		let deviceLogChildProcess;
+
+		if (launchResult) {
+			pid = launchResult.split(":")[1].trim();
+		}
+
+		if (!this.isDeviceLogOperationStarted) {
+			deviceLogChildProcess = this.getDeviceLogProcess(deviceId);
+			if (deviceLogChildProcess.stdout) {
+				deviceLogChildProcess.stdout.on("data", (data: NodeBuffer) => {
+					const dataAsString = data.toString();
+
+					if (pid) {
+						if (dataAsString.indexOf(`[${pid}]`) > -1 || dataAsString.indexOf(` ${pid} `) > -1) {
+							process.stdout.write(dataAsString);
+						}
+					} else {
+						process.stdout.write(dataAsString);
+					}
+				});
+			}
+
+			if (deviceLogChildProcess.stderr) {
+				deviceLogChildProcess.stderr.on("data", (data: string) => {
```

Comment:
The event handlers for `deviceLogChildProcess.stderr.on("data"` and `deviceLogChildProcess.stdout.on("data"` are identical - you can extract the handler as a separate function and pass it to both cases